sdks/java/io/common/src/test/java/org/apache/beam/sdk/io/common/DatabaseTestHelper.java

```diff
@@ -79,6 +80,19 @@ public static String getPostgresDBUrl(PostgresIOTestPipelineOptions options) {
         options.getPostgresDatabaseName());
   }
 
+  public static Long getPostgresTableSize(DataSource dataSource, String tableName)
+      throws SQLException {
+    try (Connection connection = dataSource.getConnection()) {
+      try (Statement statement = connection.createStatement()) {
+        ResultSet resultSet =
+            statement.executeQuery(String.format("select pg_relation_size('%s')", tableName));
+        if (resultSet.next()) {
+          return resultSet.getLong(1);
+        } else return 0L;
```

Comment:
It looks that the code returns 0L in case the size is not available (am I right?) Maybe we shouldn't return 0L in that case - "else" means in this case that "we couldn't retrieve the size" not that it is equal to 0. 

Have you considered using Optional<Long> here?

Reply:
You're right, that's a huge improvement. Thanks for the suggestion!